src/components/SearchResultItem.tsx

```diff
@@ -8,23 +8,94 @@ interface ParentTextClickedFunc {
 }
 
 interface SearchResultItemProps {
-    data: LibraryUtilities.ItemData;
-    libraryContainer: LibraryContainer;
-    highlightedText: string;
-    pathToItem: LibraryUtilities.ItemData[];
-    onParentTextClicked: ParentTextClickedFunc;
-    detailed: boolean;
+    data: LibraryUtilities.ItemData,
+    libraryContainer: LibraryContainer,
+    highlightedText: string,
+    pathToItem: LibraryUtilities.ItemData[],
+    onParentTextClicked: ParentTextClickedFunc,
+    detailed: boolean,
+    index: number
 }
 
-interface SearchResultItemStates { }
+interface SearchResultItemStates {
+    selected: boolean
+}
 
 export class SearchResultItem extends React.Component<SearchResultItemProps, SearchResultItemStates> {
 
     constructor(props: SearchResultItemProps) {
         super(props);
+        this.state = ({ selected: false });
+    }
+
+    componentWillMount() {
+        window.addEventListener("keydown", this.handleKeyDown.bind(this));
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("keydown", this.handleKeyDown.bind(this));
+    }
+
+    // Update selection state and scroll current item into view if the selected item is not in view yet.
+    componentDidUpdate() {
+        this.updateSelectionState();
+
+        if (this.state.selected) {
+            let container = ReactDOM.findDOMNode(this.props.libraryContainer);
+            let currentItem = ReactDOM.findDOMNode(this);
+            let containerRect = container.getBoundingClientRect();
+            let currentRect = currentItem.getBoundingClientRect();
+
+            if (currentRect.top < currentRect.height) {
+                currentItem.scrollIntoView();
+            }
+
+            if (currentRect.bottom > containerRect.bottom) {
+                currentItem.scrollIntoView(false);
+            }
+        }
+    }
+
+    handleKeyDown(event: any) {
+        switch (event.code) {
+            case "ArrowUp":
+                event.preventDefault(); // Prevent arrow key from navigating around search input
+                this.updateSelectionIndex(false);
+                break;
+            case "ArrowDown":
+                event.preventDefault();
+                this.updateSelectionIndex(true);
+                break;
+            case "Enter": // Allow node creation by pressing enter key
+                if (this.state.selected) {
+                    this.onItemClicked();
+                }
+            default:
+                break;
+        }
+    }
+
+    // Update the current selectionIndex in libraryContainer
+    updateSelectionIndex(selectNextItem: boolean) {
+        if (this.props.index == 0) {
```

Comment:
This is to update the selectionIndex of libraryContainer. It makes more sense for LibraryContainer to keep track of the selectionIndex, rather than every SearchResultItem has one attribute for this index. Every time when user presses arrow up/down, all the SearchResultItems would respond to the event. However, libraryContainer only needs to be called once, that's why I only make the first item set the index in libraryContainer.